### PR TITLE
docs: add a little documentation on hooks.

### PR DESCRIPTION
**Pull request by kyrofa**
Update LP: [#1586465](https://bugs.launchpad.net/ubuntu/+source/snapd/+bug/1586465) with a generic hook informational blurb.

### CODE REVIEW

**Comment by kyrofa** [comment]
> This is actually a teaser for hooks docs. Including the cliffhanger of snapctl! augh! when does season 2 come out?

Haha, yeah I suppose it is! This document will grow once the `apply-config` hook is actually usable. I'll also start a `snapctl` document discussing its various subcommands once we have one.

**Comment by mvo5** [comment]
:+1: Thanks for writing this doc!

File: docs/hooks.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+connection. These types of situations are handled by hooks.
+
+A hook is defined as an executable contained within the `meta/hooks/` directory
+inside the snap. The file name of the executable is the name of the hook (e.g.
+the upgrade hook executable would be `meta/hooks/upgrade`).
+
+As long as the file name of the executable corresponds to a supported hook name,
+that's all one needs to do in order to utilize a hook within their snap. Note
+that hooks, like apps, are executed within a confined environment. By default
+hooks will run with no plugs; if a hook needs more capabilities one can use the
```
Comment (zyga) [inline docs/hooks.md]: I'd rename "capabilities" to "permissions" as capabilities might be confused with Linux capabilities

Comment (kyrofa) [inline docs/hooks.md]: "Permissions" and "capabilities" are not synonymous in this context. I tried "privilege" instead, does that read better?

Comment (zyga) [inline docs/hooks.md]: +1

**Pull request merged**